tensorflow/python/keras/mixed_precision/experimental/autocast_variable.py

```diff
@@ -185,46 +185,66 @@ def constraint(self):
     return self._variable.constraint
 
   def assign(self, value, use_locking=None, name=None, read_value=True):
-    return self._variable.assign(value, use_locking, name, read_value)
+    assign_op = self._variable.assign(value, use_locking, name, read_value)
+    if read_value and resource_variable_ops.is_resource_variable(assign_op):
```

Comment:
Move these three lines to a shared function, named something like _maybe_wrap. Also add a comment or docstring why this is necessary.

Reply:
Good point. Made the changes in 109a41e4c324a0ebfb1206979f724d8116f916c1